src/Sarif.Driver/Sdk/AnalyzeCommandBase.cs

```diff
@@ -57,10 +57,30 @@ public string DefaultConfigurationPath
 
         public override int Run(TOptions options)
         {
-            // 0. Initialize an common logger that drives all outputs. This
-            //    object drives logging for console, statistics, etc.
-            using (AggregatingLogger logger = InitializeLogger(options))
+
+#pragma warning disable CS0618 // Type or member is obsolete
+            //  To correctly initialize the logger, we must first add Hashes to dataToInsert
+            if (options.ComputeFileHashes)
+            {
+                OptionallyEmittedData dataToInsert = options.DataToInsert.ToFlags();
+                dataToInsert |= OptionallyEmittedData.Hashes;
+
+                options.DataToInsert = Enum.GetValues(typeof(OptionallyEmittedData)).Cast<OptionallyEmittedData>()
+                    .Where(oed => dataToInsert.HasFlag(oed)).ToList();
+            }
+
+                // 0. Initialize an common logger that drives all outputs. This
+                //    object drives logging for console, statistics, etc.
+                using (AggregatingLogger logger = InitializeLogger(options))
             {
+                //  Once the logger has been correctly initialized, we can raise a warning
+                _rootContext = CreateContext(options, logger, RuntimeErrors);
+                if (options.ComputeFileHashes)
+                {
+                    Warnings.LogObsoleteOption(_rootContext, "--hashes", SdkResources.ComputeFileHashes_ReplaceInsertHashes);
+#pragma warning restore CS0618
```

Comment:
i think we should do the pragma just for the if option instead of the entire block of code.

Reply:
Were you thinking two separate pragma's?  We refer to the ComputeFileHashes property twice so we need it to cover both.

Reply:
exactly. just an style issue. otherwise, its good :)

---
In reply to: [555959576](https://github.com/microsoft/sarif-sdk/pull/2231#discussion_r555959576) [](ancestors = 555959576)

Reply:
Done